### PR TITLE
Use the absolute `abs_path` when resolving Descriptor sourceMappingURLs

### PR DESCRIPTION
**Pull request by Swatinem**
We were running into an error in `Url::parse`, because the descriptor `url` is a relative one (aka `~/foo/bar.js`) in most cases, and we need a real *absolute* Url in order to resolve the `sourceMappingURL` reference.

#skip-changelog

### CODE REVIEW

File: crates/symbolicator-service/src/services/sourcemap_lookup.rs [hunk 1]
```diff
@@ -280,6 +280,11 @@ impl SourceMapLookup {
     }
 }
 
+// TODO: Handle relative paths like `../`, `./`, etc
+fn join_url(base: &Url, url: &str) -> Option<Url> {
```
Comment (kamilogorek) [inline crates/symbolicator-service/src/services/sourcemap_lookup.rs:284]: Should we pull this util from the sentry-cli @loewenheim?

Comment (kamilogorek) [reply crates/symbolicator-service/src/services/sourcemap_lookup.rs:284]: We can also use https://docs.rs/normpath/latest/normpath/ instead to make it more battletested

Comment (loewenheim) [reply crates/symbolicator-service/src/services/sourcemap_lookup.rs:284]: I'm not sure that crate does exactly what we need here. At least in `sentry-cli`, I don't think we want to normalize leading `.`s away. Taking the function from `sentry-cli` could work, but it doesn't operate on URLs.

File: crates/symbolicator-service/src/services/sourcemap_lookup.rs [hunk 2]
```diff
@@ -280,6 +280,15 @@ impl SourceMapLookup {
     }
 }
 
+/// Joins a `url` to the `base` [`Url`], taking care of our special `~/` prefix that is treated just
+/// like an absolute url.
+fn join_url(base: &Url, url: &str) -> Option<Url> {
+    if let Some(url) = url.strip_prefix('~') {
+        return base.join(url).ok();
+    }
+    base.join(url).ok()
```
Comment (kamilogorek) [inline crates/symbolicator-service/src/services/sourcemap_lookup.rs:289]:
```suggestion
    let url = url.strip_prefix('~').unwrap_or(url);
    base.join(url).ok()
```

Comment (kamilogorek) [reply crates/symbolicator-service/src/services/sourcemap_lookup.rs:289]: (can be a oneliner as well tbh)

**Review by kamilogorek — APPROVED** [review]

**Pull request merged**